Fitness/Views/ClassDetail/ClassDetailHeaderView.swift

```diff
@@ -0,0 +1,152 @@
+//
+//  ClassDetailHeaderView.swift
+//  Fitness
+//
+//  Created by Kevin Chan on 5/18/19.
+//  Copyright © 2019 Cornell AppDev. All rights reserved.
+//
+
+import UIKit
+
+protocol ClassDetailHeaderViewDelegate: class {
+    func classDetailHeaderViewLocationSelected()
+    func classDetailHeaderViewInstructorSelected()
+}
+
+class ClassDetailHeaderView: UICollectionReusableView {
+
+    // MARK: - Private view vars
+    private let imageView = UIImageView()
+    private let imageFilterView = UIView()
+    private let semicircleImageView = UIImageView(image: #imageLiteral(resourceName: "semicircle"))
+    private let titleLabel = UILabel()
+    private let locationButton = UIButton()
+    private let instructorButton = UIButton()
+    private let durationLabel = UILabel()
+
+    // MARK: - Private data vars
+    private weak var delegate: ClassDetailHeaderViewDelegate?
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        setupViews()
+        setupConstraints()
+    }
+
+    // MARK: - Public configure
+    func configure(for delegate: ClassDetailHeaderViewDelegate, gymClassInstance: GymClassInstance) {
+        self.delegate = delegate
+        imageView.kf.setImage(with: gymClassInstance.imageURL)
+        titleLabel.text = gymClassInstance.className
+        locationButton.setTitle(gymClassInstance.location, for: .normal)
+        instructorButton.setTitle(gymClassInstance.instructor.uppercased(), for: .normal)
+        durationLabel.text = "\(Int(gymClassInstance.duration) / 60) MIN"
+    }
+
+    // MARK: - Targets
+    @objc func locationSelected() {
+        delegate?.classDetailHeaderViewLocationSelected()
+    }
+
+    @objc func instructorSelected() {
+        delegate?.classDetailHeaderViewInstructorSelected()
+    }
+
+    // MARK: - CONSTRAINTS
+    private func setupViews() {
+        imageView.contentMode = .scaleAspectFill
+        addSubview(imageView)
+
+        imageFilterView.backgroundColor = UIColor(displayP3Red: 0, green: 0, blue: 0, alpha: 0.6)
+        addSubview(imageFilterView)
+
+        semicircleImageView.contentMode = UIView.ContentMode.scaleAspectFit
+        semicircleImageView.translatesAutoresizingMaskIntoConstraints = false
+        addSubview(semicircleImageView)
+
+        titleLabel.font = ._48Bebas
+        titleLabel.textAlignment = .center
+        titleLabel.textColor = .white
+        titleLabel.sizeToFit()
```

Comment:
I feel like we can j take out all of the sizeToFit() calls but I also could be wrong

Reply:
yeah we should